### DISCUSSION TITLE
Best Practice for Combining Data from Multiple APIs into a Domain Model in Store 5

### DISCUSSION THREAD

**Discussion by 100nandoo** [dropbox/Store#675] [Q&A] [answered]
Hi,

I’m working on a use case where I need to fetch data from two separate APIs—one for media details and another for media sessions—and combine this data into a single domain model for the Media entity. Since the documentation for Store 5 is still in progress, I’m seeking guidance on the best approach to handle this scenario.

Here’s the solution I’ve come up with so far:
1. Create two separate stores:
	        * MediaDetailStore: For fetching and storing data from the Media Detail API.
	        * MediaSessionStore: For fetching and storing data from the Media Session API.
1. Combine the data from these two stores in the repository layer to build the Media domain model.

I’d like to know:
	* Is this the recommended approach for such cases?
	* Are there better practices or patterns in Store 5 for combining data from multiple sources?
	* Any additional advice or considerations I should keep in mind?

Thank you for your time, and I appreciate your insights!

**Comment by matt-ramotar** [comment, marked as answer]
Hey @100nandoo - Thanks for starting this discussion. These are good questions. I just published a guide on deciding between single or multiple stores. You can find it here: https://store.mobilenativefoundation.org/docs/best-practices/store5/single-or-multiple-stores. Let me know if it helps answers your questions
